benches/mailbox_parsing.rs

```diff
@@ -0,0 +1,22 @@
+use criterion::{black_box, criterion_group, criterion_main, Criterion};
+use lettre::message::Mailbox;
+
+fn bench_parse_single(mailbox: &str) {
+    assert!(mailbox.parse::<Mailbox>().is_ok());
+}
+
+fn criterion_benchmark(c: &mut Criterion) {
+    c.bench_function("parse single mailbox", |b| {
+        b.iter(|| bench_parse_single(black_box("Test <test@mail.local>")))
+    });
+    c.bench_function("parse multiple mailboxes", |b| {
+        b.iter(|| {
+            bench_parse_single(black_box(
```

Comment:
Shouldn't this be parsing multiple?

Side note: Maybe we should check that we consume the entire input in the Mailbox parser?

Reply:
Yes totally, I wrote the code too fast. Let me start again!

Reply:
:heavy_check_mark: